ldap/servers/slapd/back-ldbm/ldbm_search.c

```diff
@@ -1515,6 +1517,18 @@ ldbm_back_next_search_entry(Slapi_PBlock *pb)
 
     /* Find the next candidate entry and return it. */
     while (1) {
+        if (li->li_dblock_monitoring &&
+            slapi_atomic_load_32((int32_t *)&(li->li_dblock_threshold_reached), __ATOMIC_RELAXED)) {
+            errmsg = PR_smprintf("DB locks threshold is reached (nsslapd-db-locks-monitoring-threshold "
```

Comment:
From what I understand PR_smprintf might be an expensive function for this part of the code, allocating memory, formatting stuff. Could we use something like slapi_send_ldap_result(.. LDAP_DBLOCKS_EXCEEDED...)  instead.

Reply:
Good catch! Thanks!